### PR TITLE
添加api生命周期管理部分

### PR DESCRIPTION
**Pull request by caidixi**
后端添加api生命周期管理部分

### CODE REVIEW

**Comment by caidixi** [comment]
好的，收到



在 2020-04-09 09:12:46，"Zyriix" <notifications@github.com> 写道：

@Zyriix requested changes on this pull request.

格式方面稍微有些问题，文件的命名方式最好与之前的系统统一，参考文中的链接，修改一下路径

In api-server/src/main/java/com/clsaa/dop/server/api/controller/PolicyController.java:

> +    @Autowired

+    public PolicyController(PolicyService policyService) {

+        this.policyService = policyService;

+    }

+

+    @ApiOperation(value = "查看限流策略")

+    @GetMapping("/flowControl/currentLimit/get")

+    @ApiResponses({

+            @ApiResponse(code = 400,message = "错误参数")

+    })

+    public ResponseResult<CurrentLimitPolicyDetail[]> getCurrentLimitPolicy(){

+        return new ResponseResult<>(0,"success");

+    }

+

+    @ApiOperation(value = "查看限流策略详情")

+    @GetMapping("/flowControl/currentLimit/get/{policyId}")



restfulAPI中可以省略掉get/post等关键词

In api-server/src/main/java/com/clsaa/dop/server/api/dao/entity/Service.java:

> +    public Service(String id, String name, String description, Long timeout, boolean caching, Long cachingTime,

+                   boolean fuse, Long fuseDetectionRing, Long criticalFusingFailureRate, Long fuseDuration, Long replyDetectionRingSize, ServiceRoute serviceRoute) {

+        this.id = id;

+        this.name = name;

+        this.description = description;

+        this.timeout = timeout;

+        this.caching = caching;

+        this.cachingTime = cachingTime;

+        this.fuse = fuse;

+        this.fuseDetectionRing = fuseDetectionRing;

+        this.criticalFusingFailureRate = criticalFusingFailureRate;

+        this.fuseDuration = fuseDuration;

+        this.replyDetectionRingSize = replyDetectionRingSize;

+        this.serviceRoute = serviceRoute;

+    }

+



你可以安装一个lombok插件，可以用注解免去getter(),setter()的设置。参考
https://github.com/doporg/dop/blob/master/application-server/src/main/java/com/clsaa/dop/server/application/model/po/AppEnv.java 其中Data注解，Builder注解，AllArgsConstructor及Noargsconstructor注解都是使用lombok的功能

In api-server/src/main/java/com/clsaa/dop/server/api/dao/entity/Service.java:

> @@ -0,0 +1,158 @@

+package com.clsaa.dop.server.api.dao.entity;

+

+import javax.persistence.*;

+

+@Entity



我理解你这个应该是一个po model，我们一般放在model/po文件夹下

File: api-server/src/main/java/com/clsaa/dop/server/api/serviceImpl/MonitorServiceImpl.java [hunk 1]
```diff
@@ -1,32 +1,68 @@
...
+    @Override
+    public ResponseResult<TrafficStatistics> getTrafficStatistics() {
+        Date currentTime = UTCUtils.timeBefore30m();
+        TrafficStatistics trafficStatistics = new TrafficStatistics();
+        int num = 5;
+
+        //成功、失败请求数
+        List<Integer> successStatus = getSuccessHttpStatus();
+        List<Integer> failStatus = getFailHttpStatus();
+        trafficStatistics.setSuccessfulRequests(logRepository.countLogByTimeAfterAndResponseStatusIn(currentTime,successStatus));
```
Comment (Zyriix) [inline api-server/src/main/java/com/clsaa/dop/server/api/serviceImpl/MonitorServiceImpl.java:46]: 一般我们使用分页查询，可以参考https://github.com/doporg/dop/blob/master/application-server/src/main/java/com/clsaa/dop/server/application/service/ProjectService.java以及https://github.com/doporg/dop/blob/master/application-server/src/main/java/com/clsaa/dop/server/application/dao/ProjectRepository.java

File: api-server/src/main/java/com/clsaa/dop/server/api/serviceImpl/MonitorServiceImpl.java [hunk 2]
```diff
@@ -81,4 +117,103 @@ public ResponseResult receiveRequestLog(KongHttpLog kongHttpLog) {
...
+                    apiSimpleInfo.setResponseTime(responseTime);
+                    apiSimpleInfos.add(apiSimpleInfo);
+                    if (apiSimpleInfos.size()>=num){
+                        break;
+                    }
+                }
+                i++;
+            }
+        }
+        return apiSimpleInfos;
```
Comment (Zyriix) [inline api-server/src/main/java/com/clsaa/dop/server/api/serviceImpl/MonitorServiceImpl.java:194]: 格式转化有空添加一下 BO层到VO层 使用之前说的Beanutils

File: api-server/src/main/java/com/clsaa/dop/server/api/serviceImpl/MonitorServiceImpl.java [hunk 3]
```diff
@@ -81,4 +117,103 @@ public ResponseResult receiveRequestLog(KongHttpLog kongHttpLog) {
...
+    }
+
+    private List<ApiSimpleInfo> getFailService(Date timestamp,int num){
+        List<ApiSimpleInfo> apiSimpleInfos = new LinkedList<>();
+        List<Object[]> results = logRepository.findFailService(timestamp,getFailHttpStatus());
+        if (results.size()>0&&results.get(0).length>0){
+            int i = 0;
+            while (i<results.size()){
+                String serviceId = String.valueOf(results.get(i)[0]);
```
Comment (Zyriix) [inline api-server/src/main/java/com/clsaa/dop/server/api/serviceImpl/MonitorServiceImpl.java:203]: 这个地方可以批量拿取结果，传入数组返回数组，可以参考https://github.com/doporg/dop/blob/master/application-server/src/main/java/com/clsaa/dop/server/application/dao/ProjectRepository.java  line：63 FindallbyIdin()

**Pull request merged**